modules/measure/src/main/java/com/opengamma/strata/measure/fxopt/FxOptionVolatilitiesMarketDataFunction.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2017 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.measure.fxopt;
+
+import static com.opengamma.strata.collect.Guavate.toImmutableList;
+
+import java.time.LocalDate;
+import java.time.ZonedDateTime;
+import java.util.stream.IntStream;
+
+import com.google.common.collect.ImmutableList;
+import com.opengamma.strata.basics.ReferenceData;
+import com.opengamma.strata.calc.marketdata.MarketDataConfig;
+import com.opengamma.strata.calc.marketdata.MarketDataFunction;
+import com.opengamma.strata.calc.marketdata.MarketDataRequirements;
+import com.opengamma.strata.collect.array.DoubleArray;
+import com.opengamma.strata.data.scenario.MarketDataBox;
+import com.opengamma.strata.data.scenario.ScenarioMarketData;
+import com.opengamma.strata.measure.ValuationZoneTimeDefinition;
+import com.opengamma.strata.pricer.fxopt.FxOptionVolatilities;
+import com.opengamma.strata.pricer.fxopt.FxOptionVolatilitiesId;
+
+/**
+ * Market data function that builds FX option volatilities.
+ * <p>
+ * This function creates FX option volatilities, turning {@code FxOptionVolatilitiesId} into {@code FxOptionVolatilities}.
+ */
+public class FxOptionVolatilitiesMarketDataFunction
+    implements MarketDataFunction<FxOptionVolatilities, FxOptionVolatilitiesId> {
+
+  @Override
+  public MarketDataRequirements requirements(FxOptionVolatilitiesId id, MarketDataConfig marketDataConfig) {
+
+    FxOptionVolatilitiesDefinition volatilitiesDefinition = marketDataConfig.get(
+        FxOptionVolatilitiesDefinition.class, id.getName().getName());
```

Comment:
Also obtain a `ValuationZoneTimeDefinition` so that the code fails here rather than in `build` if it is missing. This will need a comment to explain what is going on!

Reply:
It seems that our approach is to log failure during building market data rather than building dependency tree.